### PR TITLE
Corner case validations added for Juniper MX

### PR DESCRIPTION
**Pull request by stephanerobert**
Some more test and bugfixes for the Juniper MX

### CODE REVIEW

**Comment by stephanerobert** [comment]
Travis fails because we need to bump Fake-Switches

File: netman/adapters/switches/juniper/mx.py [hunk 1]
```diff
@@ -467,6 +472,12 @@ def parse_vrrp_groups(self, interface_unit):
     def parse_icmp_redirects(self, interface_unit):
         return first(interface_unit.xpath("family/inet/no-redirects")) is None
 
+    def get_delete_vlan_element(self, mode):
```
Comment (lindycoder) [inline netman/adapters/switches/juniper/mx.py]: That's 2 methods

File: tests/adapters/switches/juniper_mx_test.py [hunk 2]
```diff
@@ -2098,6 +2098,289 @@ def test_set_access_vlan_on_interface_that_already_has_it_does_nothing(self):
...
+            <config>
+              <configuration>
+                <interfaces>
+                  <interface>
+                    <name>xe-0/0/6</name>
+                    <unit>
+                      <name>0</name>
+                      <family>
+                        <bridge>
+                          <vlan-id operation="delete"/>
```
Comment (lindycoder) [inline tests/adapters/switches/juniper_mx_test.py]: you don't really need this statement, does work though...  but not if the operation=delete comes after the vlan-id, then it will remove... i don't really like that we depend on the order of the statements in an XML structure
